pkgs/development/tools/rust/cargo-llvm-cov/default.nix

```diff
@@ -1,36 +1,59 @@
 { stdenv
 , lib
-, fetchCrate
+, fetchurl
+, fetchFromGitHub
 , rustPlatform
+, rustc
 }:
 
-rustPlatform.buildRustPackage rec {
+let
   pname = "cargo-llvm-cov";
   version = "0.5.25";
 
-  src = fetchCrate {
-    inherit pname version;
-    sha256 = "sha256-4ctwvDLluJsLWJPInoFGqxmEzlEuBtEJb3/x+q/5pDA=";
+  owner = "taiki-e";
+  homepage = "https://github.com/${owner}/${pname}";
+
+  llvm = rustc.llvmPackages.llvm;
+
+  # Download `Cargo.lock` from crates.io so we don't clutter up Nixpkgs
+  cargoLock = fetchurl {
+    name = "Cargo.lock";
+    url = "https://crates.io/api/v1/crates/${pname}/${version}/download";
+    sha256 = "sha256-6VLAno4XhThzf8zdxWo32tfaHvOpHGOFwoDxMfm9ONs=";
+    downloadToTemp = true;
+    postFetch = ''
+      tar xzf $downloadedFile ${pname}-${version}/Cargo.lock
+      mv ${pname}-${version}/Cargo.lock $out
+    '';
   };
+in
+
+rustPlatform.buildRustPackage {
+  inherit pname version;
+
+  # Use `fetchFromGitHub` instead of `fetchCrate` because the latter does not
+  # pull in fixtures needed for the test suite
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-oBbGAU99kurEqbzv+PLXnDrvm9lQisYddwRSy//2bVE=";
+
+    # Upstream doesn't include the lockfile so we need to add it back
+    postFetch = ''
```

Comment:
might be better to put this in `postFetch` of the outer derivation

Reply:
this wasn't possible when the PR was created, might it should be good now thanks to #221093

Reply:
Thanks for spelling it out for me in https://github.com/NixOS/nixpkgs/pull/197478#pullrequestreview-1574403652 since I wasn't really sure what you were asking for lol. I've just applied this suggestion.